File: src/sync/js/heartbeat.js
import { EventBus } from "../../eventbus/js/eventBus.js";
import { Telemetry } from "../../telemetry/js/telemetry.js";

export class Heartbeat {
  constructor(interval = 5000) {
    this.interval = interval;
    this.bus = new EventBus();
    this.telemetry = new Telemetry();
    this.timer = null;
  }

  start() {
    this.timer = setInterval(() => {
      const pulse = {
        timestamp: Date.now(),
        pulse: "alive"
      };
      this.bus.emit("heartbeat", pulse);
      this.telemetry.log("heartbeat", pulse);
    }, this.interval);
  }

  stop() {
    clearInterval(this.timer);
  }
}

File: src/sync/python/heartbeat.py
import time
import threading
from eventbus.python.event_bus import EventBus
from telemetry.python.telemetry import Telemetry

class Heartbeat:
    def __init__(self, interval=5):
        self.interval = interval
        self.running = False
        self.bus = EventBus()
        self.telemetry = Telemetry()

    def start(self):
        self.running = True
        threading.Thread(target=self._loop, daemon=True).start()

    def _loop(self):
        while self.running:
            payload = {"timestamp": time.time(), "pulse": "alive"}
            self.bus.emit("heartbeat", payload)
            self.telemetry.log("heartbeat", payload)
            time.sleep(self.interval)

    def stop(self):
        self.running = False
